pkg/cloudprovider/aws/factory.go

```diff
@@ -87,9 +89,15 @@ func (f *Factory) CapacityFor(provisioner *v1alpha1.Provisioner) cloudprovider.C
 }
 
 func withRegion(sess *session.Session) *session.Session {
-	region, err := ec2metadata.New(sess).Region()
-	log.PanicIfError(err, "failed to call the metadata server's region API")
-	sess.Config.Region = aws.String(region)
+	region, envIsSet := os.LookupEnv("AWS_REGION")
```

Comment:
I think if you use AWS_DEFAULT_REGION, it will be overridden automatically. 

https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html

Reply:
Seems like the safe way would be to look for `AWS_DEFAULT_REGION` and use that if set, otherwise fall back to `AWS_REGION`. Adding this behavior.